static/app/views/performance/landing/widgets/widgets/lineChartListWidgetMetrics.tsx

```diff
@@ -0,0 +1,202 @@
+import {Fragment, useMemo, useState} from 'react';
+import {withRouter} from 'react-router';
+
+import _EventsRequest from 'sentry/components/charts/eventsRequest';
+import Count from 'sentry/components/count';
+import Truncate from 'sentry/components/truncate';
+import {t} from 'sentry/locale';
+import MetricsRequest from 'sentry/utils/metrics/metricsRequest';
+import {decodeList} from 'sentry/utils/queryString';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import useApi from 'sentry/utils/useApi';
+import _DurationChart from 'sentry/views/performance/charts/chart';
+import {transactionSummaryRouteWithQuery} from 'sentry/views/performance/transactionSummary/utils';
+
+import {excludeTransaction} from '../../utils';
+import {GenericPerformanceWidget} from '../components/performanceWidget';
+import SelectableList, {
+  GrowLink,
+  ListClose,
+  RightAlignedCell,
+  Subtitle,
+  WidgetEmptyStateWarning,
+} from '../components/selectableList';
+import {transformMetricsToArea} from '../transforms/transformMetricsToArea';
+import {transformMetricsToList} from '../transforms/transformMetricsToList';
+import {PerformanceWidgetProps, QueryDefinition, WidgetDataResult} from '../types';
+import {PerformanceWidgetSetting} from '../widgetDefinitions';
+
+type DataType = {
+  chart: WidgetDataResult & ReturnType<typeof transformMetricsToArea>;
+  list: WidgetDataResult & ReturnType<typeof transformMetricsToList>;
+};
+
+const framesList = [
+  PerformanceWidgetSetting.MOST_SLOW_FRAMES,
+  PerformanceWidgetSetting.MOST_FROZEN_FRAMES,
+];
+
+// This file is simpler than singleFieldAreaWidget because metrics for now support only MOST_SLOW_FRAMES and MOST_FROZEN_FRAMES
+export function LineChartListWidgetMetrics(props: PerformanceWidgetProps) {
+  const api = useApi();
+  const [selectedListIndex, setSelectListIndex] = useState(0);
+  const {
+    ContainerActions,
+    organization,
+    fields,
+    chartSetting,
+    location,
+    chartColor,
+    chartHeight,
+  } = props;
+  const field = fields[0];
+
+  if (fields.length !== 1) {
+    throw new Error(`Line chart list widget can only accept a single field (${fields})`);
+  }
+
+  if (!framesList.includes(chartSetting)) {
```

Comment:
do we need this extra check here?

Reply:
not necessarily, I would prefer to leave it there to be super explicit on the component level (at least until we support more chartSettings)